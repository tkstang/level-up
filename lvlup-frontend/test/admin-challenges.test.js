import AdminChallenges from '../src/components/Admin/challenges/admin-challenges/challenges-container';
import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import lvlupApp from '../src/reducers/index';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

test('AdminChallenges should render', () => {
  const store = mockStore({ lvlupApp });
  const adminchallenges = shallow(
    <AdminChallenges store={store} />,
  );
  expect(shallowToJson(adminchallenges)).toMatchSnapshot();
});
