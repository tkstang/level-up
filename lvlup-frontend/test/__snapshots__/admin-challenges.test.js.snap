exports[`test AdminChallenges should render 1`] = `
<lifecycle(ChallengesTable)
  campusChallenges={[Function]}
  makeChallengeInactive={[Function]}
  resetAddChallenge={[Function]}
  resetChallengeList={[Function]}
  resetEditChallenge={[Function]}
  selectChallenge={[Function]}
  sortChallengeCategory={[Function]}
  sortChallengeCategoryReverse={[Function]}
  sortChallengeName={[Function]}
  sortChallengeNameReverse={[Function]}
  sortChallengePoints={[Function]}
  sortChallengePointsReverse={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
  submissionsAction={[Function]} />
`;
